go/core/action.go

```diff
@@ -27,9 +27,11 @@ import (
 // stream the results by invoking the callback periodically, ultimately returning
 // with a final return value. Otherwise, it should ignore the StreamingCallback and
 // just return a result.
-type Func[In, Out, Stream any] func(context.Context, In, func(context.Context, Stream) error) (Out, error)
+type Func[In, Out, Stream any] = func(context.Context, In, func(context.Context, Stream) error) (Out, error)
 
-// TODO: use a generic type alias for the above when they become available?
+// Middleware is a function that wraps an action execution, similar to HTTP middleware.
+// It can modify the input, output, and context, or perform side effects.
+type Middleware[In, Out, Stream any] = func(ctx context.Context, input In, cb func(context.Context, Stream) error, next Func[In, Out, Stream]) (Out, error)
```

Comment:
This middleware signature is similar to js (express) middleware and I'm OK with it, but in Go a more idiomatic signature might be:

```go
type Middleware[I, O, S any] func(Func[I, O, S]) Func[I, O, S]
```

Similar to https://pkg.go.dev/go.ntrrg.dev/ntgo/net/http/middleware

I had it previously proposed in https://github.com/firebase/genkit/pull/429/files#diff-181ec49997af46351ae6b830a6c8883400c96f71628a4198a988e23b7d461d09

I'm OK either way, just asking if you considered it?